api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -18,6 +18,33 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Wasm]
 // [#extension: envoy.bootstrap.wasm]
 
+// Configuration for restricting proxy-wasm capabilities available to modules.
+message CapabilityRestrictionConfig {
+  // The proxy-wasm capabilities which will be allowed. Capabilities are mapped by
+  // name. The *SanitizationConfig* which each capability maps to is currently unimplemented and ignored,
+  // and so should be left empty.
+  //
+  // The capability names are given in the
+  // `proxy-wasm ABI <https://github.com/proxy-wasm/spec/tree/master/abi-versions/vNEXT>`_.
+  // Additionally, the following WASI capabilities from
+  // `this list <https://github.com/WebAssembly/WASI/blob/master/phases/snapshot/docs.md#modules>`_
+  // are implemented and can be allowed:
+  // *fd_write*, *fd_read*, *fd_seek*, *fd_close*, *fd_fdstat_get*, *environ_get*, *environ_sizes_get*,
+  // *args_get*, *args_sizes_get*, *proc_exit*, *clock_time_get*, *random_get*.
+  map<string, SanitizationConfig> allowed_capabilities = 1;
```

Comment:
should this be `capability_configs`, or something like that? The content is not necessarily `allowlist` and sometimes `denylist`, isn't it

Reply:
same goes for proxy-wasm-cpp-host side PR

Reply:
The keys in the map are always allowed capabilities, so I think the name is still accurate. The `SanitizationConfig` which each capability maps to contains either an allowlist or denylist of arguments, but the capability itself is always allowed.

Reply:
ah, right. Thanks